src/buildlog/mod.rs

```diff
@@ -51,10 +52,46 @@ impl BuildLog {
 		}
 	}
 
+	pub fn add_entry(&mut self, output: RawString, build_starttime: Instant, starttime: Instant, endtime: Instant, command: RawString) {
+		self.entries.insert(output, Entry {
+			start_time_ms: (starttime - build_starttime).as_millis() as u32,
+			end_time_ms:   (endtime - build_starttime).as_millis() as u32,
+			restat_mtime:  0 /* TODO */,
+			command_hash:  murmur_hash_64a(command.as_bytes()),
+		});
+	}
+
 	pub fn read(file: impl AsRef<Path>) -> Result<BuildLog, Error> {
 		BuildLog::read_from(File::open(file)?)
 	}
 
+	pub fn write(&self, file: impl AsRef<Path>) -> Result<(), Error> {
+		// TODO: should we append to this file instead of truncating it?
+		self.write_to(File::create(file)?)
+	}
+
+	pub fn write_to(&self, file: File) -> Result<(), Error> {
+		let mut file = BufWriter::new(file);
+
+		file.write(b"# ninja log v5\n")?;
```

Comment:
`.write` doesn't necessarily write all bytes. It returns how many bytes were written. You probably want to use [`write_all`](https://doc.rust-lang.org/beta/std/io/trait.Write.html#method.write_all) instead.